### PR TITLE
Reduce test suite run time

### PR DESCRIPTION
**Pull request by ShaneHarvey**
Helps with #528. Shaves off almost 20 minutes from the test suite ([22 minutes](https://travis-ci.org/mongodb-labs/mongo-connector/builds/160289698) compared to [39 minutes](https://travis-ci.org/ShaneHarvey/mongo-connector/builds/155408276)).

### CODE REVIEW

File: tests/test_filter_fields.py [hunk 1]
```diff
@@ -42,10 +51,8 @@ def tearDown(self):
         try:
             self.opman.join()
         except RuntimeError:
-            pass                # OplogThread may not have been started
-        self.primary_conn.drop_database("test")
-        close_client(self.primary_conn)
-        self.repl_set.stop()
+            # OplogThread may not have been started
+             pass
```
Comment (llvtt) [inline tests/test_filter_fields.py]: Looks like the comment/`pass` might be indented at different levels.

File: tests/test_filter_fields.py [hunk 2]
```diff
@@ -22,15 +22,16 @@
 from mongo_connector.doc_managers.doc_manager_simulator import DocManager
 from mongo_connector.locking_dict import LockingDict
 from mongo_connector.oplog_manager import OplogThread
-from mongo_connector.test_utils import ReplicaSet, assert_soon, close_client
+from mongo_connector.test_utils import ReplicaSetSingle, assert_soon, \
+    close_client
```
Comment (llvtt) [inline tests/test_filter_fields.py]: You can use `()` to do imports that span multiple lines, which pep8 recommends over the backslash.

File: mongo_connector/test_utils.py [hunk 3]
```diff
@@ -170,9 +171,39 @@ def start(self):
         return self._init_from_response(self._make_post_request())
 
 
+class ReplicaSetSingle(MCTestObject):
```
Comment (llvtt) [inline mongo_connector/test_utils.py]: Seems like this could just extend ReplicaSet and override get_config.

File: mongo_connector/test_utils.py [hunk 4]
```diff
@@ -202,11 +235,15 @@ def start(self):
...
         self.id = response['id']
         self.uri = response.get('mongodb_auth_uri', response['mongodb_uri'])
-        self.shards = [ReplicaSet()._init_from_response(resp)
+        self.shards = [self._shard_type()._init_from_response(resp)
                        for resp in (shard1, shard2)]
         return self
 
 
+class ShardedClusterSingle(ShardedCluster):
+    _shard_type = ReplicaSetSingle
```
Comment (llvtt) [inline mongo_connector/test_utils.py]: Nice

File: mongo_connector/test_utils.py (hunk 3 above)
Comment (ShaneHarvey) [inline mongo_connector/test_utils.py]: Right, it's much simpler now.

**Comment by ShaneHarvey** [comment]
Thanks, I'll hold the urge to change unrelated lines of code from now on.

**Pull request merged**